docs/r1cs-docs-example.md

```diff
@@ -25,31 +29,26 @@ For K > 1:
     (x_{k-2} - z)---⊗      ⊗---(y_{k-2} - z)    // mulx[k-2], muly[k-2]
                    /        \
     (x_{k-1} - z)_/          \_(y_{k-1} - z)
+```
 
-	Connect left and right sides of the shuffle statement
-	    mulx_out[0] = muly_out[0]
+Connect the left and right sides of the shuffle statement:
+    `mulx_out[0] = muly_out[0]`
 
-	For i == [0, k-3]:
-	    mulx_left[i]  = x_i - z
-	    mulx_right[i] = mulx_out[i+1]
-	    muly_left[i]  = y_i - z
-	    muly_right[i] = muly_out[i+1]
+For i == [0, k-3]:
+    `mulx_left[i]  = x_i - z`
+    `mulx_right[i] = mulx_out[i+1]`
+    `muly_left[i]  = y_i - z`
+    `muly_right[i] = muly_out[i+1]`
 
-	The last multipliers connect two last variables (on each side)
-	    mulx_left[k-2]  = x_{k-2} - z
-	    mulx_right[k-2] = x_{k-1} - z
-	    muly_left[k-2]  = y_{k-2} - z
-	    muly_right[k-2] = y_{k-1} - z
+The last multipliers connect the two last variables (on each side)
+    `mulx_left[k-2]  = x_{k-2} - z`
+    `mulx_right[k-2] = x_{k-1} - z`
+    `muly_left[k-2]  = y_{k-2} - z`
+    `muly_right[k-2] = y_{k-1} - z`
 
 For K = 1:
-	Connect x to y directly, omitting the challenge entirely as it cancels out
-    	x_0 = y_0
-```
-
-This shuffle `prove` function takes a list of scalar inputs and outputs, makes commitments to them, and creates a proof that the committed outputs are a valid reordering of the committed inputs. 
-The shuffle `verify` function verifies that, given a shuffle proof and a list of committed inputs and outputs, the outputs are a valid reordering of the inputs.
-
-Because only the prover knows the scalar values of the inputs and outputs, and the verifier only sees the (blinded) committed inputs and outputs, the verifier has no knowledge of what the underlying inputs and output values are. Therefore, the only information the verifier learns from this protocol is whether or not the committed outputs are a valid shuffle of the committed inputs - this is why it is a zero-knowledge proof.
+Connect x to y directly, omitting the challenge entirely as it cancels out
```

Comment:
It might be good to note that there's only one permutation of a 1-element list (so it makes sense that the challenges cancel out)